trailmap/pathway_analysis.py

```diff
@@ -0,0 +1,116 @@
+
+    plt.figure()
+    # nx.draw_shell(G, with_labels=True)
+    nx.draw_networkx_nodes(G, pos)
+    nx.draw_networkx_labels(G, pos)
+    nx.draw_networkx_edges(G, pos)
+    # nx.draw_networkx_edge_labels(G, pos)
+
+    return plt
+
+
+def draw_path(G, path):
+    draw_graph(G)
+    pos = nx.drawing.nx_pydot.graphviz_layout(G)
+    nx.draw_networkx_nodes(G, pos=pos, nodelist=path, node_color="red")
+    plt.show()
+    return
+
+
+def print_graph_parameters(G):
+    '''Prints a set of parameters characterizing the graph
+    '''
+    print('\nGraph parameters')
+
+    semiconnected = nx.is_semiconnected(G)
+    print('Is the graph semiconnected? ' + str(semiconnected))
+    if semiconnected is False:
+        print("-->This is likely because you have multiple source facilities")
+
+    hierarchy = nx.flow_hierarchy(G)
+    print("Graph hierarchy is " + str(hierarchy))
+
+    return
+
+
+def find_node_disjoint_paths(G, s, t):
+    ndp = set()
+    if s in G and t in G:
+        paths = list(nx.node_disjoint_paths(G, s, t))
+        [ndp.add(tuple(path)) for path in paths]
+
+    return ndp
+
+
+def find_maximum_flow(G, s, t):
+    '''Requires edge attribute 'capacity'
+    '''
+    max_flow_path = maximum_flow(G, s, t)
+    max_flow = maximum_flow_value(G, s, t)
+    return max_flow_path, max_flow
+
+
+def find_simple_cycles(G):
+    sc = list(nx.simple_cycles(G))
+    return sc
+
+
+def find_facility_specific_paths(pathways, facility, type='any'):
+    '''finds pathways that include a given facility. Options: 'beginning',
+    'end', 'any' specify whether the facility should be the start, the end, or
+    anywhere in the path
+    '''
+    subset_pathways = []
+    for path in pathways:
+        if facility in path:
+            if type == 'start' and path[0] == facility:
+                subset_pathways.append(path)
+            elif type == 'end' and path[-1] == facility:
+                subset_pathways.append(path)
+            else:
+                subset_pathways.append(path)
+
+    return subset_pathways
+
+
+def find_paths_between_facilities(pathways, source, target):
+    subset_pathways = []
+    for path in (x for x in pathways if x[0] == source and x[-1] == target):
+        subset_pathways.append(path)
+
+    return subset_pathways
+
+
+def find_paths_containing_facilities(pathways, facilities, type='any'):
+    p = []
+    if type == 'any':
+        for path in pathways:
+            p.append([path for facility in facilities if facility in path])
+    if type == 'all':
+        for path in pathways:
+            if set(facilities).issubset(path) is True:
+                p.append(path)
+
+    return p
+
+
+def find_shortest_path(pathways):
+    '''finds the pathway with the shortest number of steps from source to
+    target. Returns a tuple with path and length.
+    '''
+    if len(pathways) is not 0:
+        shortest_path_length = min([len(path) for path in pathways])
+        shortest_path = min([path for path in pathways])
```

Comment:
Have you tested this?  I'm not sure it works the way you think it does...

Also, while this is a compact notation, it requires you to loop over pathways twice instead of once.

Reply:
```suggestion
        shortest_path = [path for path in pathways if len(path) == shortest_length]
```

Reply:
pathway analysis has been taken out of this PR, but I did make that change. Thanks!